onnx2tf/ops/ConvTranspose.py

```diff
@@ -35,76 +33,81 @@ def make_node(
     input_tensor = get_constant_or_variable(
-        graph_node.inputs[0],
+        graph_node_input,
         before_op_output_shape_trans,
     )
+    input_tensor = tf_layers_dict[input_tensor.name]['tf_node'] \
+        if isinstance(input_tensor, gs.Variable) else input_tensor
+    input_tensor_shape = input_tensor.shape
+    graph_node_input_shape = graph_node_input.shape
+    input_tensor_rank = len(input_tensor_shape)
+    spatial_size = input_tensor_rank - 2
+
+    # ONNX weight input
     kernel_shape = graph_node.attrs.get('kernel_shape', [])
-    kernel_size = len(kernel_shape)
     input_weights = get_weights_constant_or_variable(
         const_or_var=graph_node.inputs[1],
-        kernel_size=kernel_size,
+        kernel_size=len(kernel_shape),
     )
+    input_weights = tf_layers_dict[input_weights.name]['tf_node'] \
+        if isinstance(input_weights, gs.Variable) else input_weights
+
+    # ONNX bias input
     input_bias = None
     if len(graph_node.inputs) >= 3:
         input_bias = get_constant_or_variable(
             graph_node.inputs[2],
             before_op_output_shape_trans,
             is_bias=True,
         )
-    graph_node_output: gs.Variable = graph_node.outputs[0]
-    output_tensor_shape = graph_node_output.shape
-    dtype = graph_node_output.dtype
+        input_bias = tf_layers_dict[input_bias.name]['tf_node'] \
+            if isinstance(input_bias, gs.Variable) else input_bias
 
-    input_tensor = tf_layers_dict[input_tensor.name]['tf_node'] \
-        if isinstance(input_tensor, gs.Variable) else input_tensor
-    input_tensor_shape = input_tensor.shape
-    input_weights = tf_layers_dict[input_weights.name]['tf_node'] \
-        if isinstance(input_weights, gs.Variable) else input_weights
-    input_bias = tf_layers_dict[input_bias.name]['tf_node'] \
-        if isinstance(input_bias, gs.Variable) else input_bias
-
-    input_tensor_shape = input_tensor.shape
-    input_tensor_rank = len(input_tensor_shape)
-    spatial_size = input_tensor_rank - 2
-    input_weights_shape = input_weights.shape
-    auto_pad = graph_node.attrs.get('auto_pad', 'NOTSET')
-    dilations = graph_node.attrs.get('dilations', [1] * spatial_size)
-    group = graph_node.attrs.get('group', 1)
-    output_padding = graph_node.attrs.get('output_padding', None)
-    output_shape = graph_node.attrs.get('output_shape', None)
     pads = graph_node.attrs.get('pads', [0, 0] * spatial_size)
     strides = graph_node.attrs.get('strides', [1] * spatial_size)
+    dilations = graph_node.attrs.get('dilations', [1] * spatial_size)
 
-    # Preserving Graph Structure (Dict)
-    tf_layers_dict[graph_node_output.name] = {
-        'optype': graph_node.op,
-        'shape': output_tensor_shape,
-        'dtype': dtype,
-        'nhwc': True,
-    }
+    # get ONNX convolution output shape
+    graph_node_output_shape = graph_node.attrs.get('output_shape', graph_node_output.shape)
+    assert graph_node_output_shape is not None
```

Comment:
why would graph_node_output.shape ever be None?

Reply:
`graph_node_output` gets the shape of the red arrowhead. However, the shape of this part (in the example `[1,8,32,32]`) will not be written to the ONNX file until after the shape estimation process in ONNX (`onnx.shape_inference.infer_shapes(model)` [ONNX Shape Inference](https://github.com/onnx/onnx/blob/main/docs/ShapeInference.md)) or the shape fixing process by `onnxsim`.
![image](https://user-images.githubusercontent.com/33194443/207737531-63b39f23-27af-43be-83a3-cc0be03f184a.png)

e.g.
![image](https://user-images.githubusercontent.com/33194443/207738321-252e280c-5dae-46e4-8a57-3078ac2ea5da.png)

I do not own the ONNX file you are using for your test, so I cannot investigate the exact reason. However, one possible reason for the always None is likely that the batch size, channel, H, or W of the input OP has an undefined dimension, and the shape estimation fails at the beginning of the onnx2tf model optimization process.

e.g.
![image](https://user-images.githubusercontent.com/33194443/207738722-c6c2bfb7-0427-4220-9d40-42c9aac247a6.png)

The above guess is only an estimate based on the one sentence you reported, so I know almost nothing about it.

Reply:
ok, so if onnxsim runs, there will always be a shape then?

Reply:
Yes, I believe that the shape will be set. I believe that the shape is almost always set unless under special conditions.

Reply:
Do you plan to commit any additional revisions? If you don't have any additional commits planned, I'm planning to merge them in, since all the CI tests have passed.

Reply:
I should probable remove the assert then

Reply:
done

Reply:
OK. Thanks!